### PR TITLE
enhancement/486/exportLayers-tabs

### PR DESCRIPTION
**Pull request by jrdrgz**
still waiting on some design feedback and suggestions. (#486)

### CODE REVIEW

**Comment by cwilmsen** [comment]
Some notes:

Code
-You are using double quotes in some areas, but not in others (I prefer not using any)
-You are using '==' and '!=', and not '===' and '!=='
-export-picker.js; 
--Line 208: It would be more consistent to use 'exportRadio.prop('checked', false);'
--jquery object variables should start with '$' ie '$exportRadio' and not 'exportRadio'
--Line 17: Missing ';'
--Line 102: extra line
--$modalDialog is never declared as a variable
--Line 306: missing ';' (and class instead of manual setting of width)
--showExtraOptions(): should there be 2 animations on $exportContainer? They do almost exactly the same thing (especially the setting of opactiy)
--Try not to use $var.css('prop', 'something'), but instead use classes. Having a class for 'opaque:1' will make it easier to change file-wide, easier to add/remove, and easier to check for truth ( if($var.hasClass('not-opaque'){} )
-export-picker.css
--Line 15: should probably just be 'padding:0'
--extra line at the bottom
-export-picker.hbs
--Lines 9-12: extra lines
--Lines 12-55: not lined up correctly
-fake-extension.js, Line 145: 'paramns.queryId' should be 'params.queryId', and there is an extra line there
-Ditto for mock-extension.js
-mock-configuration: Before the actual merge, enable kml and remove googlemaps
-mock-extrension
--Line 16: datasource is never used
-csv-extension.js
--Line 26: extra line
--Lines 29-39: trash
-main.css
--Lines 333+: don't need them

UI
-I can't get back to extra options after I click on layers
-I can't get back to export options after clicking on layers twice
-'selectAll' doesn't turn off after unselecting a layer
-The single point mode is very scrunched. I think there needs to be padding on the left of export radios?
-Person stuff
--I think in single point mode, selecting an export option with extra options moves the modal too much. It grows and shrinks a lot, which makes it hard to follow during transitions
--Having ':' after 'Export Options' and 'Additional Options' looks funny
--The text field feels too small. Try typing in something that you would put into a similar field. If you use all lower case letters, it fits 20 characters, but larger letters fills it up very fast.
--Not sure how to deal with it, but it looks funny having column headers larger/bolder than the modal title. Maybe just make the divs a different color?

I know a lot of this is nitpicky, but I figured I'd just get it all in there.

**Comment by cwilmsen** [comment]
More stuff:
-Before merging, we need to make google maps only work for mock single points
-'select all' starts as checked, but does not actually select all the layers
I can try to export invalid options:
1. a mock and a fake query on the map
2. open export picker
3. select the fake layer
4. select KML
5. select the mock layer
(the kml option goes away)
6. You can click export
-There are a lot of issues with export not being disabled when it should be
-'select all' is checked when exporting from timeline

**Comment by cwilmsen** [comment]
-On single point opening, the modal gets big, then gets small again before the user does anything

**Comment by cwilmsen** [comment]
the googlemap lat/lon are backwards when sending to Google Maps?

**Comment by jrdrgz** [comment]
For tracking purposes, merge ticket #523 before this one. It's a fix that was done in this branch.
